### PR TITLE
[MF-180] Factor out mocks to core tooling package

### PR DESCRIPTION
**Pull request by samuelmale**
Ticket: https://issues.openmrs.org/browse/MF-180

### CODE REVIEW

File: packages/esm-devtools-app/jest.config.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = {
   moduleNameMapper: {
     "\\.(css)$": "identity-obj-proxy",
     "@openmrs/esm-framework":
-      "<rootDir>/__mocks__/openmrs-esm-framework.mock.tsx",
+      "@openmrs/esm-framework/mocks/openmrs-esm-framework.mock.tsx",
```
Comment (FlorianRappl) [inline packages/esm-devtools-app/jest.config.js:9]: Would a module name like `@openmrs/esm-framework/mock` work? Assuming that we place `mocks/openmrs-esm-framework.mock.tsx` in `mock.tsx`.

File: packages/esm-framework/mocks/react-i18next.mock.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+const React = require("react");
```
Comment (FlorianRappl) [inline packages/esm-framework/mocks/react-i18next.mock.js:1]: Hm why is this file in there? It's not part of the framework.

**Review by FlorianRappl — COMMENTED** [review]
Goes in the right direction, but we should work on the dev experience. The module name / path should be as simple as possible.

Also only make a mock for the framework. Don't provide mocks for anything else.

File: packages/esm-devtools-app/jest.config.js (hunk 1 above)
Comment (samuelmale) [reply packages/esm-devtools-app/jest.config.js:9]: > Assuming that we place mocks/openmrs-esm-framework.mock.tsx in mock.tsx.

@FlorianRappl I don't understand what you mean here.

Comment (samuelmale) [reply packages/esm-devtools-app/jest.config.js:9]: You want me to rename `mocks/openmrs-esm-framework.mock.tsx` to `mocks/mock.tsx`?

Comment (FlorianRappl) [reply packages/esm-devtools-app/jest.config.js:9]: yes

Comment (FlorianRappl) [reply packages/esm-devtools-app/jest.config.js:9]: See original request

> "@openmrs/esm-framework": "@openmrs/esm-framework/mock"

File: packages/esm-framework/mocks/react-i18next.mock.js (hunk 2 above)
Comment (brandones) [reply packages/esm-framework/mocks/react-i18next.mock.js:1]: Yeah, but it is unfortunately necessary to mock it this way in all of the MFEs. I'd advocate including it.

Comment (FlorianRappl) [reply packages/esm-framework/mocks/react-i18next.mock.js:1]: Why?

We have other things that qualify, too, in that area - so for consistency reasons I'd prefer to keep the framework package to framework mocks. This should not end up as a trash repository to just bring anything, but things that are esm-framework related. `react-i18next` is not.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]
Please keep it limited to the framework mock.

File: packages/esm-framework/mocks/react-i18next.mock.js (hunk 2 above)
Comment (samuelmale) [reply packages/esm-framework/mocks/react-i18next.mock.js]: Removed react-i18next.mock.js file.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]
Still missing:

- reportError
- interpolateString
- navigate
- getCurrentUser
- Extension
- ConfigurableLink


I got that list only by looking at some of the esm-framework.mock files in the other microfrontends.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks 🍻 !

**Pull request merged**